packages/search-service/src/search-mapping/schema.ts

```diff
@@ -0,0 +1,31 @@
+import { Document, Model, model, Schema } from 'mongoose';
+
+export const SearchMapSchema: Schema = new Schema({
+  name: String,
+  fields: [{
+    from: String,
+    to: String
+  }],
+  apiConfig: {
+    model: { type: String, enum: ['GRAPHQL', 'REST'] },
+    mode: String,
+    apiUrl: String,
+    param: String,
+    gqlQuery: String
```

Comment:
Add support for user defined Authorization header, and other custom headers if their API has some special header requirements to function properly.

Also make the necessary changes in the helper functions.
```suggestion
    gqlQuery: String,
    authorizationHeader: { type: String, },
    customHeaders: [{ key: String, value: String, }],
```